evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -133,22 +133,23 @@ open class PackageRule(
         }
     }
 
-    fun issue(severity: Severity, message: String) = issue(severity, pkg.id, null, null, message)
+    fun issue(severity: Severity, message: String, howToFix: String) =
+        issue(severity, pkg.id, null, null, message, howToFix)
 
     /**
      * Add a [hint][Severity.HINT] to the list of [issues].
      */
-    fun hint(message: String) = hint(pkg.id, null, null, message)
+    fun hint(message: String, howToFix: String) = hint(pkg.id, null, null, message, howToFix)
 
     /**
      * Add a [warning][Severity.WARNING] to the list of [issues].
      */
-    fun warning(message: String) = warning(pkg.id, null, null, message)
+    fun warning(message: String, howToFix: String) = warning(pkg.id, null, null, message, howToFix)
```

Comment:
Just thinking out loud: if the code firing the violations start needing to care about (how to fix and message)
maybe the rule implementations become more bloated. In particular if we add more stuff in.

An alternative would be that we have a component which allows resolving howToFix string given a rule violation ID. That component I'd rather see more on the UI side then.
One good thing is that UI change (adjusting hints) is likely frequent.
But those frequent changes would not go into the rules code which imo is desirable as 
the rules are more critical than the UI.

...probably my comment would make more sense if the rules were not in KTS but I like to bring it up still.

Reply:
Since rules are custom code written by the user I think the best place to define the "how to fix" text is next to the rule itself.
In general I would prefer if the user could define a template for the message to not have it in the model at all, but only apply the template in the reporter, but that's much more work.

Reply:
After thinking about it again, I have an idea - not sure how strong that is though - what do you think about this:

1. Keep the signature of the issue functions as is
2. adjust the internal implementation of those issue function to resolve the hints themselves by e.g. a hint resolver which might get the params: rule-id, package-id.
3. put such hint resolver implementation still into kts as a compromise.

This would separate hint resolution quite a bit - well the caller of this e.g. warning API could also use such resolver but that would lead to a bit redundant / ugly code of the callers of warning. This is why I was saying in the previous review cycle that it'd be nice to have those functions not unused because that'd show how the caller would look like.

What's your opinion on that? Imo would be nice to have also yours, @sschuberth  ?!

Reply:
I still think from the user perspective it is much more straightforward to define the how to fix message as part of the rule instead of in a separate place.